### PR TITLE
Migrate InformationTablesTableInfo to SystemTable

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

This also changes `tableParameters` from a `Map<String, Object>` to
`Settings`. That avoids some copy operations when we create the
`DocTableInfo` instances and also makes it easier to reason about the
types of the values.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by mfussenegger** [comment]
@seut @kovrus 

Tests on this PR currently fail because `information_schema.tables` now no longer returns `null` for `settings` in system tables, but a map that has the full shape (keys for all child columns) - with all leaf values being `null`.

I could add a hook to override that behavior to have `null` at the root level, but I'm not sure if that is actually better. WDYT?

**Comment by seut** [comment]
> I could add a hook to override that behavior to have null at the root level, but I'm not sure if that is actually better. WDYT?

I like your approach to avoid special cases for system tables, so I'd be fine with the change.
On the other hand, this is some kind of a breaking change, I think it is worth it, but we may have to document it.

**Comment by kovrus** [comment]
+1 as well for keeping the full shape objects with null values

**Comment by mfussenegger** [comment]
I decided to keep the behavior with the `null` for now. We can follow up on it in a separate PR if we want to, to split the behavior change from refactoring of internals.

File: sql/src/test/java/io/crate/integrationtests/InformationSchemaTest.java [hunk 1]
```diff
@@ -601,7 +601,7 @@ public void testTableConstraintsWithOrderBy() {
     @Test
     public void testDefaultColumns() {
         execute("select * from information_schema.columns order by table_schema, table_name");
-        assertEquals(772, response.rowCount());
+        assertEquals(774, response.rowCount());
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/InformationSchemaTest.java:604]: just curious: what caused this change?

Comment (mfussenegger) [reply sql/src/test/java/io/crate/integrationtests/InformationSchemaTest.java:604]: `settings.translog.durability` and `settings.scheduler.max_merge_count` were returned by the expressions but not defined in the schema.

**Pull request merged**